crates/walrus-service/src/node/shard_sync.rs

```diff
@@ -32,8 +32,9 @@ enum SyncShardResult {
     /// The shard sync finished successfully.
     Success,
     /// The shard sync is not finished and should be retried after a backoff.
-    /// The bool indicates whether to directly recover the shard instead of using shard sync.
-    RetryAfterBackoff(bool),
+    /// The first bool indicates whether to directly recover the shard instead of using shard sync.
+    /// The second bool indicates whether the shard sync made progress.
+    RetryAfterBackoff(bool, bool),
```

Comment:
Suggestion: Since there are two values here, perhaps use a named enum variant.

```suggestion
    /// The first bool indicates whether to directly recover the shard instead of using shard sync.
    /// The second bool indicates whether the shard sync made progress.
    RetryAfterBackoff{force_recovery: bool, shard_sync_made_progress: bool},
````

Reply:
Looks better! Thanks!